DDG4/edm4hep/EDM4hepFileReader.cpp

```diff
@@ -0,0 +1,320 @@
+    podio::Frame frame = m_reader.readEntry("events", event_number);
+    const auto& primaries = frame.get<edm4hep::MCParticleCollection>(m_collectionName);
+    int eventNumber = event_number, runNumber = 0;
+    if (primaries.isValid()) {
+      //Read the event header collection and add it to the context as an extension
+      const auto& eventHeaderCollection = frame.get<edm4hep::EventHeaderCollection>(m_eventHeaderCollectionName);
+      if(eventHeaderCollection.isValid() && eventHeaderCollection.size() == 1){
+        const auto& eh = eventHeaderCollection.at(0);
+        eventNumber = eh.getEventNumber();
+        runNumber = eh.getRunNumber();
+        try {
+          Geant4Context* ctx = context();
+          ctx->event().addExtension<edm4hep::MutableEventHeader>(new edm4hep::MutableEventHeader(eh.clone()));
+        } catch(std::exception& ) {}
+        // Create input event parameters context
+        try {
+          Geant4Context* ctx = context();
+          EventParameters *parameters = new EventParameters();
+          parameters->setRunNumber(runNumber);
+          parameters->setEventNumber(eventNumber);
+          parameters->ingestParameters(frame.getParameters());
+          ctx->event().addExtension<EventParameters>(parameters);
+        } catch(std::exception &) {}
+      } else {
+        printout(WARNING,"EDM4hepFileReader","No EventHeader collection found or too many event headers!");
+        try {
+          Geant4Context* ctx = context();
+          EventParameters *parameters = new EventParameters();
+          parameters->setRunNumber(0);
+          parameters->setEventNumber(event_number);
+          parameters->ingestParameters(frame.getParameters());
+          ctx->event().addExtension<EventParameters>(parameters);
+        } catch(std::exception &) {}
+      }
+      printout(INFO,"EDM4hepFileReader","read collection %s from event %d in run %d ",
+               m_collectionName.c_str(), eventNumber, runNumber);
+
+    } else {
+      return EVENT_READER_EOF;
+    }
+
+    printout(INFO,"EDM4hepFileReader", "We read the particle collection");
+    int NHEP = primaries.size();
+    // check if there is at least one particle
+    if ( NHEP == 0 ) {
+      printout(WARNING,"EDM4hepFileReader", "We have no particles");
+      return EVENT_READER_NO_PRIMARIES;
+    }
+
+    MCPARTICLE_MAP mcparts;
+    std::vector<int>  mcpcoll;
+    mcpcoll.resize(NHEP);
+    for(int i=0; i<NHEP; ++i ) {
+      edm4hep::MCParticle p = primaries.at(i);
+      mcparts[p.getObjectID().index] = i;
+      mcpcoll[i] = p.getObjectID().index;
+    }
+
+    // build collection of MCParticles
+    for(int i=0; i<NHEP; ++i )   {
+      const auto& mcp = primaries.at(i);
+      Geant4ParticleHandle p(new Particle(i));
+      const auto mom   = mcp.getMomentum();
+      const auto vsx   = mcp.getVertex();
+      const auto vex   = mcp.getEndpoint();
+      const auto spin  = mcp.getSpin();
+      const auto color = mcp.getColorFlow();
```

Comment:
Slated for removal. https://github.com/key4hep/EDM4hep/pull/389

Reply:
I think since we plan to remove it in any case from EDM4hep, it doesn't make too much sense to add it here in the first place.

Reply:
removed colorflow.